tests/unit/src/Internal/Service/PaymentContextLoggerServiceTest.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Class ContextLoggerServiceTest
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Tests\Internal\Service;
+
+use WCPAY_UnitTestCase;
+use WCPay\Internal\Logger;
+use WCPay\Internal\Payment\Change;
+use WCPay\Internal\Payment\PaymentContext;
+use WCPay\Internal\Payment\Transition;
+use WCPay\Internal\Service\PaymentContextLoggerService;
+use WCPay\Internal\Payment\PaymentMethod\NewPaymentMethod;
+
+/**
+ * Level3 data service unit tests.
+ */
+class PaymentContextLoggerServiceTest extends WCPAY_UnitTestCase {
+	/**
+	 * Service under test.
+	 *
+	 * @var PaymentContextLoggerService
+	 */
+	private $sut;
+
+	/**
+	 * @var Logger|MockObject
+	 */
+	private $mock_logger;
+
+	/**
+	 * @var PaymentContext|MockObject
+	 */
+	private $mock_payment_context;
+
+	/**
+	 * Set up the tests.
+	 */
+	protected function setUp(): void {
+		parent::setUp();
+		$this->mock_logger = $this->createMock( Logger::class );
+		$this->sut         = new PaymentContextLoggerService( $this->mock_logger );
+
+		$this->mock_payment_context = $this->getMockBuilder( PaymentContext::class )
+			->disableOriginalConstructor()
+			->onlyMethods( [ 'get_transitions', 'get_order_id' ] )
+			->getMock();
+		$this->mock_payment_context
+			->method( 'get_order_id' )
+			->will( $this->returnValue( 123 ) );
+	}
+
+	protected function setup_transitions() {
+		return [
+			new Transition( strtotime( '2023-10-26 00:00:00' ), null, 'Initial_State', [ new Change( 'key_1', null, 'new_value_1' ) ] ),
+			new Transition(
+				strtotime( '2023-10-26 00:00:10' ),
+				'Initial_State',
+				'Final_State',
+				[
+					new Change( 'key_1', 'old_value_1', 'new_value_1' ),
+					new Change( 'key_2', null, 'new_value_2' ),
+					new Change( 'key_3', 'old_value_3', 'new_value_3' ),
+				]
+			),
+			new Transition( strtotime( '2023-10-26 00:00:40' ), 'Final_State', null, [ new Change( 'key_1', 'old_value_4', 'new_value_4' ) ] ),
+		];
+	}
+
+	/**
+	 * Test the basic happy path of logging a payment context with changes
+	 */
+	public function test_log_changes() {
+		$this->mock_payment_context
+			->method( 'get_transitions' )
```

Comment:
Let's add the `->expects()` call, as my IDE is complaining that mock objects don't have a `method` method.
```suggestion
		$this->mock_payment_context->expects( $this->once() )
			->method( 'get_transitions' )
```

Reply:
Done